doc/api/index.rst

```diff
@@ -1,67 +1,45 @@
-API Overview
-============
+API
+===
 
 .. toctree::
    :hidden:
 
    api_changes
 
-.. contents:: :local:
+For recent changes, see :doc:`api_changes`.
 
-See also the :doc:`api_changes`.
+When using the library you will typically create
+:doc:`Figure <figure_api>` and :doc:`Axes <axes_api>` objects and 
+call their methods to add content and modify the apprearance.
 
-Usage patterns
---------------
-
-Below we describe several common approaches to plotting with Matplotlib.
-
-The pyplot API
-^^^^^^^^^^^^^^
-
-`matplotlib.pyplot` is a collection of command style functions that make
-Matplotlib work like MATLAB. Each pyplot function makes some change to a
-figure: e.g., creates a figure, creates a plotting area in a figure, plots
-some lines in a plotting area, decorates the plot with labels, etc.
+  - :doc:`figure_api`: axes creation, figure-level content
+  - :doc:`axes_api`: most plotting methods, Axes labels, access to 
+    axis styling, etc.  
 
-`.pyplot` is mainly intended for interactive plots and simple cases of
-programmatic plot generation.
+Example: We create a Figure ``fig`` and Axes ``ax``. Then we call
+methods on them to plot data, add axis labels and a figure title.
+ 
+.. plot::
+   :include-source:
+   :align: center
 
-Further reading:
+   import matplotlib.pyplot as plt
+   import numpy as np
 
-- The `matplotlib.pyplot` function reference
-- :doc:`/tutorials/introductory/pyplot`
-- :ref:`Pyplot examples <pyplots_examples>`
+   x = np.arange(0, 4, 0.1)
+   y = np.sin(x*np.pi)
 
-.. _api-index:
+   fig, ax = plt.subplots(figsize=(3,2), constrained_layout=True)
+   ax.plot(x, y)
+   ax.set_xlabel('t [s]')
+   ax.set_ylabel('S [V]')
+   fig.suptitle('Sine wave')
 
-The object-oriented API
-^^^^^^^^^^^^^^^^^^^^^^^
-
-At its core, Matplotlib is object-oriented. We recommend directly working
-with the objects, if you need more control and customization of your plots.
-
-In many cases you will create a `.Figure` and one or more
-`~matplotlib.axes.Axes` using `.pyplot.subplots` and from then on only work
-on these objects. However, it's also possible to create `.Figure`\ s
-explicitly (e.g. when including them in GUI applications).
-
-Further reading:
-
-- `matplotlib.axes.Axes` and `matplotlib.figure.Figure` for an overview of
-  plotting functions.
-- Most of the :ref:`examples <examples-index>` use the object-oriented approach
-  (except for the pyplot section)
-
-The pylab API (disapproved)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-.. automodule:: pylab
-   :no-members:
 
 Modules
 -------
 
-Matplotlib consists of the following submodules:
+Alphabetical list of submodules:
 
 .. toctree::
    :maxdepth: 1
```

Comment:
```suggestion
   :maxdepth: 1
   :multicol-toc:
```
Maybe.
![grafik](https://user-images.githubusercontent.com/2836374/111854685-c1e68180-8920-11eb-9568-f5ec9cc7b9c3.png)

Reply:
Is this unreleased?  I get `/Users/jklymak/matplotlib/doc/api/index.rst:45: WARNING: Error in "toctree" directive:
unknown option: "multicol-toc".` on sphinx 3.5.3

Reply:
Sorry, this should have been `:class: multicol-toc`.

`multicol-toc` is just a CSS class we have added to our `mpl.css`.

Reply:
oops reached the internet - didn't think to search our source code...

Reply:
Sorry this does not seem to work. `.. toctree:` does not support a `:class:` attribute.

I've otherwise only used it for `.. contents:` so far. The above screenshot was a hacked CSS in the browser, because I didn't want to wait for a docs build.

Seems we have to live with the long list for now.

Reply:
Ha!  I thought I was going crazy...  Certainly adding a class directive to toctree seems like a good idea, but seems like an upstream issue.